File: my_py_pkg/my_py_pkg/my_first_node.py
#!usr/bin/env python3
import rclpy
from rclpy.node import Node


class MyFirstNode(Node):
    def __init__(self):
        super().__init__("my_first_node")
        self._counter = 0
        self.get_logger().info("Hello ROS2")
        self.timer = self.create_timer(1.0, self.timer_callback)

    def timer_callback(self):
        self._counter += 1
        self.get_logger().info(f"Timer callback triggered. Count: {self._counter}")


def main(args=None):
    rclpy.init(args=args)

    node = MyFirstNode()
    rclpy.spin(node)

    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<rclcpp::Node>("my_first_node");
    RCLCPP_INFO(node->get_logger(), "Hello, ROS2 from C++17");
    
    rclcpp::spin(node);
    
    rclcpp::shutdown();
    return 0;
}